test/units/testsuite-04.sh

```diff
@@ -141,4 +141,49 @@ sleep 3
 # https://github.com/systemd/systemd/issues/15528
 journalctl --follow --file=/var/log/journal/*/* | head -n1 || [[ $? -eq 1 ]]
 
+function set_unit_log_regexes() {
+    UNIT=$1
+    LOG_FILTER=${2:-""}
+
+    mkdir -p /etc/systemd/system/"$UNIT".d/
+    echo "[Service]" > /etc/systemd/system/logs-filtering.service.d/override.conf
+    echo "LogFilterRegex=$LOG_FILTER" >> /etc/systemd/system/logs-filtering.service.d/override.conf
+    systemctl daemon-reload
+}
+
+function run_service_and_fetch_logs() {
+    UNIT=$1
+
+    START=$(date '+%Y-%m-%d %T.%6N')
+    systemctl restart "$UNIT"
+    sleep .5
+    journalctl --sync
+    END=$(date '+%Y-%m-%d %T.%6N')
+
+    journalctl -q -u "$UNIT" -S "$START" -U "$END" | grep -Pv "systemd\[[0-9]+\]"
+    systemctl stop "$UNIT"
+}
+
+# Accept all log messages
+set_unit_log_regexes "logs-filtering.service" ""
+[[ -n $(run_service_and_fetch_logs "logs-filtering.service") ]]
```

Comment:
My bash fu is pretty horrible so could you describe how this works exactly?

Reply:
`run_service_and_fetch_logs "logs-filtering.service"` will run `logs-filtering.service` a few seconds until the unit has enough time to send logs to systemd-journald. Then it will fetch logs from this unit from journalctl and print them.

`[[ -n $(run_service_and_fetch_logs "logs-filtering.service") ]]` is a shortcut for `test -n $(run_service_and_fetch_logs "logs-filtering.service")` which means: if the length of the string (which is what the subcommand will print, i.e. journalctl's output) is non zero, the `test` returns 0 (the test succeeded). 

Basically, for `[[ -n ...]]`, we expect journalctl to have logs from the service, for `[[ -z ... ]]` (with `-z` meaning the length of the following string **is** 0, we expect journalctl to not have logs from the service.

Reply:
What if any of the commands other than `journalctl` produce output? Wouldn't that interfere with the test?

Reply:
If other command from `run_service_and_fetch_logs` produce output, that would interfere, yes.